### DISCUSSION TITLE
Building the editor from the ground-up

### DISCUSSION THREAD

**Discussion by andyjessop** [toeverything/blocksuite#5321] [General]
Just started with Blocksuite, and it looks really incredible so far. Although, I admit I'm feeling a little lost when it comes to customising. I want to be able to do the following:

- build out a skeleton LitElement component with just, say, text blocks. No images, no database, etc.
- tree-shake the bundle so a minimum is delivered (I believe the `simple-affine-editor` is about 4MB.

Are there any guidelines on this in the docs? Or an example in the wild?

Edit: to add to this, it would be great if there were a tutorial that would walk you through setting up a basic editor from scratch.

**Comment by doodlewind** [comment]
@andyjessop thanks for your interest! Currently the editors are [composed in this way](https://github.com/toeverything/blocksuite/blob/master/packages/blocks/src/_presets/index.ts#L94) so I think technically you can define your own one referring to this example. Since we've just published the new docs and the editor API surface, it would require some more time providing new corresponding examples (and tailored builds), PR welcome surely. 

Anyway, customizing capabilities you mentioned is very well aligned with what we have planned, hope you can stay tuned!

**Reply by andyjessop** [reply]
Thanks for the quick reply, much appreciated. I get the feeling this lib is exactly what I need for my project (simple/local note-taking), but the docs are a little too sparse right now. I'll definitely stay tuned though!

**Reply by andyjessop** [reply]
Btw, I recommend [this](https://diataxis.fr/) as essential reading before embarking on writing documentation. It will help you keep it structured (organising docs is _hard_).
